esphome/components/at581x/binary_sensor.py

```diff
@@ -0,0 +1,27 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import binary_sensor
+from esphome.const import (
+    CONF_HAS_TARGET,
+    DEVICE_CLASS_MOTION,
+    ICON_MOTION_SENSOR,
+)
+
+from . import CONF_AT581X_ID, AT581XComponent
+
+DEPENDENCIES = ["at581x"]
+
+CONFIG_SCHEMA = {
+    cv.GenerateID(CONF_AT581X_ID): cv.use_id(AT581XComponent),
+    cv.Optional(CONF_HAS_TARGET): binary_sensor.binary_sensor_schema(
+        device_class=DEVICE_CLASS_MOTION,
+        icon=ICON_MOTION_SENSOR,
+    ),
+}
```

Comment:
> I'm not 100% sure I get it here. This radar detects motion not static target (unlike 24GHz or 60GHz? IIUC). If you stop moving in front of the radar, the detection signal will go down.
> 
> In the datasheet, there's also a light detection sensor, but I don't know if it's an error in the Chinese to English translation (in that case, light would mean RF emission, I'm not sure), or if it's actually here.
>
> What's sure is that on Box S3, the radar is under a opaque plastic cover so it can't detect light.
>
> What do you mean about not making a child sensor, do you have an example I can look at?

So because this is the only binary sensor, there is no real need to have it as a deeper object in the config. See the below suggestion.

```suggestion
CONFIG_SCHEMA = (
    binary_sensor.binary_sensor_schema(
        device_class=DEVICE_CLASS_MOTION,
        icon=ICON_MOTION_SENSOR,
    ).extend(cv.Schema(
        {
            cv.GenerateID(CONF_AT581X_ID): cv.use_id(AT581XComponent),
        }
    )
)

```

This would allow config like:
```yaml
binary_sensor:
  - platform: at581x
    name: Motion Detected
    at581x_id: ...
```